### PR TITLE
Switch to nginx-unprivileged base image to not run as root.

### LINKED ISSUE #396: the renku-ui container should not run as root user
The container should run as non-root user.

### PR DESCRIPTION
**Pull request by ciyer**
Fix #396

### CODE REVIEW

**Comment by rokroskar** [comment]
did you try deploying this? I assume you need to change the ports in the chart: https://github.com/SwissDataScienceCenter/renku-ui/blob/master/helm-chart/renku-ui/templates/deployment.yaml#L21

**Comment by ciyer** [comment]
I thought I had deployed this, but it turns out that I had deployed a different version.

File: Dockerfile [hunk 1]
```diff
@@ -10,12 +10,20 @@ COPY src /app/src/
...
 COPY --from=0 /app/build /usr/share/nginx/html
 COPY nginx.vh.default.conf /etc/nginx/conf.d/default.conf
 COPY docker-entrypoint.sh /app/docker-entrypoint.sh
 
+# Set up the config file written by docker-entrypoint
+USER root
+RUN touch /usr/share/nginx/html/config.json
+RUN chmod a+r /usr/share/nginx/html/config.json
+RUN chown 1001 /usr/share/nginx/html/config.json
+USER 1001
```
Comment (jirikuncar) [inline Dockerfile]: we should make sure we store the user id before switching to root

File: helm-chart/renku-ui/values.yaml [hunk 2]
```diff
@@ -54,7 +54,7 @@ image:
 
 service:
   type: ClusterIP
-  port: 80
+  port: 8080
```
Comment (jirikuncar) [inline helm-chart/renku-ui/values.yaml]:
```suggestion
  nodePort: 80
  port: 8080
```

Comment (rokroskar) [reply helm-chart/renku-ui/values.yaml]: Are you sure you mean `nodePort`? Or just `targetPort`?

Comment (jirikuncar) [reply helm-chart/renku-ui/values.yaml]: It should have been a `targetPort`.

Comment (jirikuncar) [reply helm-chart/renku-ui/values.yaml]: see https://github.com/SwissDataScienceCenter/renku-ui/blob/master/helm-chart/renku-ui/templates/service.yaml#L14

File: Dockerfile (hunk 1 above)
Comment (ciyer) [reply Dockerfile]: I changed to using the user name specified in the nginx base image.

File: Dockerfile [hunk 3]
```diff
@@ -10,12 +10,20 @@ COPY src /app/src/
...
 COPY --from=0 /app/build /usr/share/nginx/html
 COPY nginx.vh.default.conf /etc/nginx/conf.d/default.conf
 COPY docker-entrypoint.sh /app/docker-entrypoint.sh
 
+# Set up the config file written by docker-entrypoint
+USER root
+RUN touch /usr/share/nginx/html/config.json
+RUN chmod a+r /usr/share/nginx/html/config.json
+RUN chown nginx /usr/share/nginx/html/config.json
+USER nginx
```
Comment (jirikuncar) [inline Dockerfile]: 👍

File: helm-chart/renku-ui/values.yaml (hunk 2 above)
Comment (ciyer) [reply helm-chart/renku-ui/values.yaml]: I have been trying to get this to work, but I cannot. I do prefer this suggestion, but I feel that enough time has been spent trying to get it to work for now. I recommend switching to port 8080 and revisiting this later.

**Pull request merged**